pkg/rbac/sharedData.go

```diff
@@ -51,6 +55,82 @@ var managedClusterResourceGvr = schema.GroupVersionResource{
 	Resource: "managedclusters",
 }
 
+func (shared *SharedData) getPropertyTypes(cache *Cache, ctx context.Context) (map[string]string, error) {
+	propTypeMap := make(map[string]string)
+
+	// original query: select distinct key, jsonb_typeof(value) as datatype FROM search.resources,jsonb_each(data);
+	var selectDs *goqu.SelectDataset
+
+	//define schema:
+	schemaTable := goqu.S("search").Table("resources")
+
+	//data expression to get value and key
+	jsb := goqu.L("jsonb_each(?)", goqu.C("data"))
+
+	//select from these datasets
+	ds := goqu.From(schemaTable, jsb)
+
+	//select statement with orderby and distinct clause
+	selectDs = ds.Select(goqu.L("key"), goqu.L("jsonb_typeof(?)",
+		goqu.C("value")).As("datatype")).Distinct()
+
+	query, params, err := selectDs.ToSQL()
+	if err != nil {
+		klog.Errorf("Error building Search query: %s", err.Error())
+		return propTypeMap, err
+	}
+
+	klog.V(5).Infof("Query for property datatypes: [%s] ", query)
+	rows, err := cache.pool.Query(ctx, query, params...)
+	if err != nil {
+		klog.Errorf("Error resolving query [%s] with args [%+v]. Error: [%+v]", query, err)
+		return propTypeMap, err
+	}
+	defer rows.Close()
+	for rows.Next() {
+		var key, value string
+		err = rows.Scan(&key, &value)
+
+		if err != nil {
+			klog.Errorf("Error %s scanning value for getPropertyTypes:%s", err.Error(), query)
+			continue
+		}
+		propTypeMap[key] = value
+
+	}
+	//cache query:
+	shared.propTypes = propTypeMap
+	shared.propTypeErr = err
+	shared.propTypeTime = time.Now()
+
+	return propTypeMap, err
+}
+
+func (cache *Cache) GetPropertyTypes(ctx context.Context, refresh bool) (map[string]string, error) {
+
+	// propTypesMap := make(map[string]string)
+	//check if propTypes data in cache and not nil and return
+	if len(cache.shared.propTypes) > 0 && cache.shared.propTypeErr == nil && !refresh {
+		klog.V(6).Info("Using property types from cache.")
+
+		propTypesMap := cache.shared.propTypes
+		return propTypesMap, nil
+
+	} else {
+		//run query to refresh data
+		propTypes, err := cache.shared.getPropertyTypes(cache, ctx)
+		if err != nil {
```

Comment:
Move this log to reduce how much data we log.
```suggestion
	if len(cache.shared.propTypes) > 0 && cache.shared.propTypeErr == nil && !refresh {
		propTypesMap := cache.shared.propTypes
		return propTypesMap, nil
	} else {
		// run query to refresh data
		klog.V(6).Info("Getting property types from database.")
		propTypes, err := cache.shared.getPropertyTypes(cache, ctx)
		if err != nil {
```